test/unit/loops.spec.ts

```diff
@@ -754,6 +754,142 @@ test("forof forwarded lua iterator with tupleReturn", () => {
     expect(result).toBe("0a1b2c");
 });
 
+test("forof array empty destructuring declaration", () => {
```

Comment:
Declaration/assignment versions of each test take quite a lot of space, maybe it would be better to generate them if the only difference is `const []`/`[]`? Something like:

```ts
describe("for...of empty destructuring", () => {
    const declareTests = (destructuringPrefix: string) => {
        test("array", () => {
            const code = `
                const arr = [["a"], ["b"], ["c"]];
                let i = 0;
                for (${destructuringPrefix}[] of arr) {
                    ++i;
                }
                return i;
            `;
            expect(util.transpileAndExecute(code)).toBe(3);
        });

        // ...
    };

    describe("declaration", () => declareTests("const "));
    describe("assignment", () => declareTests(""));
});
```